docs/src/FAQ.md

```diff
@@ -76,3 +76,93 @@ ChainRulesTestUtils.jl has some dependencies, so it is a separate package from C
 This means your package can depend on the light-weight ChainRulesCore.jl, and make ChainRulesTestUtils.jl a test-only dependency.
 
 Remember to read the section on [On writing good `rrule` / `frule` methods](@ref).
+
+## How do chain rules work for complex functions?
+
+ChainRules follows the convention that `frule` applied to a function ``f(x + i y) = u(x,y) + i v(x,y)`` with perturbation ``\Delta x + i \Delta y`` returns the value and
+```math
+\tfrac{\partial u}{\partial x} \, \Delta x + \tfrac{\partial u}{\partial y} \, \Delta y + i \, \Bigl( \tfrac{\partial v}{\partial x} \, \Delta x + \tfrac{\partial v}{\partial y} \, \Delta y \Bigr)
+.
+```
+Similarly, `rrule` applied to the same function returns the value and a pullback function which, when applied to the adjoint ``\Delta u + i \Delta v``, returns
+```math
+\Delta u \, \tfrac{\partial u}{\partial x} + \Delta v \, \tfrac{\partial v}{\partial x} + i \, \Bigl(\Delta u \, \tfrac{\partial u }{\partial y} + \Delta v \, \tfrac{\partial v}{\partial y} \Bigr)
+.
+```
+If we interpret complex numbers as vectors in ``\mathbb{R}^2``, then `frule` (`rrule`) corresponds to multiplication with the (transposed) Jacobian of ``f(z)``, i.e. `frule` corresponds to
+```math
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} \, \Delta x + \tfrac{\partial u}{\partial y} \, \Delta y
+\\
+\tfrac{\partial v}{\partial x} \, \Delta x + \tfrac{\partial v}{\partial y} \, \Delta y
+\end{pmatrix}
+=
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}
+\begin{pmatrix}
+\Delta x \\ \Delta y
+\end{pmatrix}
+
+```
+and `rrule` corresponds to
+```math
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} \, \Delta u + \tfrac{\partial v}{\partial x} \, \Delta v
+\\
+\tfrac{\partial u}{\partial y} \, \Delta u + \tfrac{\partial v}{\partial y} \, \Delta v
+\end{pmatrix}
+=
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}^T
+\begin{pmatrix}
+\Delta u \\ \Delta v
+\end{pmatrix}
```

Comment:
I think there's a missing minus sign here in the `∂/∂y` terms when we compare to https://github.com/JuliaDiff/ChainRulesCore.jl/pull/167/files#diff-a5f548edfc6b320f451aa0189937f0e8R138-R146. 

`imag(du_dz)` is equal to `-du_dy`, not `+du_dy`.

Reply:
Actually, after checking my math, I think it's this section that's okay and `jacobian_via_rrule` that needs minus signs on the `imag` parts.

Reply:
Actually, after checking my math, I think it's this section that's okay and `jacobian_via_rrule` that needs minus signs on the `imag` parts.

Reply:
Okay, actually none of this is wrong, it's just that I think when @ettersi wrote 
```julia
_,du_dz = pullback( 1)
_,dv_dz = pullback(im)
```
he was referring to `du_dz` as `∂u/∂z = ∂u/∂x + i ∂u/∂y` rather than `∂u/∂z = ∂u/∂x - i ∂u/∂y`. 

Perhaps these should be renamed `du_dzconj` and `dv_dzconj` or something just in case someone takes that too literally? It's not a big deal though and maybe this is only confusing to me.

Reply:
> Perhaps these should be renamed `du_dzconj` and `dv_dzconj` or something like
> 
> ```julia
> du_dx, du_dy = reim(pullback( 1)[2])
> dv_dx, du_dy = reim(pullback(im)[2])
> ```
> 
> just in case someone takes the `d_dz` too literally? It's not a big deal though and maybe this is only confusing to me.

I second the `reim` approach for clarity.

Reply:
Agreed, that was confusing. Thanks for spotting and fixing this!